Serenity-Website/src/Sign-up/SignUp.jsx

```diff
@@ -37,34 +51,50 @@ function SignUp() {
     setPassword(e.target.value);
   };
 
+  const handleChangeConfirmPassword = (e) => {
+    setConfirmPassword(e.target.value);
+  };
+
   const backendUrl = import.meta.env.VITE_BACKEND_ADDRESS;
   const url = `${backendUrl}/create`;
 
   const handleCreate = async (e) => {
     e.preventDefault();
     e.stopPropagation();
 
-    if (email && password) {
+    if (email && password && confirmPassword) {
       const isEmailValid = validateEmail(email);
       const isPasswordValid = password.length >= 8;
+      const isConfirmPasswordValid = password === confirmPassword;
 
       // update error messages if both password and email are invalid
-      if (!isEmailValid && !isPasswordValid) {
+      if (!isEmailValid && !isPasswordValid && !isConfirmPasswordValid) {
         setEmailError("Invalid email address");
         setPasswordError("Password must be at least 8 characters long");
+        setConfirmPasswordError("Passwords do not match!");
         return;
       }
 
-      // If email is invalid, notify user
-      setEmailError(isEmailValid ? null : "Invalid email address!");
+      if (!isEmailValid || !isPasswordValid || !isConfirmPasswordValid) {
+        // If email is invalid, notify user
+        if (!isEmailValid) {
+          setEmailError("Invalid email address!");
+        }
 
-      //If password length is less than required, notify user
-      setPasswordError(
-        isPasswordValid ? null : "Password must be at least 8 characters long!"
-      );
+        //If password length is less than required, notify user
+        if (!isPasswordValid) {
+          setPasswordError("Password must be at least 8 characters long!");
+        }
+
+        // if confirm password does not match, notify user
+        if (!isConfirmPasswordValid) {
+          setConfirmPasswordError("Passwords do not match!");
+        }
+        return;
+      }
 
       // Return if either validation fails
-      if (!(isEmailValid && isPasswordValid)) {
+      if (!(isEmailValid && isPasswordValid && isConfirmPasswordValid)) {
```

Comment:
Is this needed still? Looks like this case is handled above on lines 78-93

Reply:
Right! I just took it out